sympy/functions/elementary/piecewise.py

```diff
@@ -512,7 +512,11 @@ def handler(ipw):
                 cond = (x < b)
             else:
                 cond = (x <= b)
-            args.append((sum, cond))
+            if sum.has(S.Infinity, S.NegativeInfinity, S.ComplexInfinity, Undefined):
```

Comment:
Two notes:
1) 
```python
from sympy.polys.polyutils import illegal  # [nan, oo, -oo, zoo] but import at top
sum.has(*illegal)
```
2) this may only be correct if there is a single tuple in done. If there are more then the proper change needs to happen were sum is being accumulated. And I'm not exactly sure what has to be done but suspect it needs to be done there. Please check.

Reply:
;-) maybe `illegal` should be named `nanoozoo`

Reply:
> Two notes: 1)
> 
> ```python
> from sympy.polys.polyutils import illegal  # [nan, oo, -oo, zoo] but import at top
> sum.has(*illegal)
> ```
Yes, using this sounds like a good idea. I'll add it in.
> 
>     2. this may only be correct if there is a single tuple in done. If there are more then the proper change needs to happen were sum is being accumulated. And I'm not exactly sure what has to be done but suspect it needs to be done there. Please check.
I'm not sure what exactly you mean here. I tried my code with cases where `done` has more than one tuple in it and it seems to be working fine.